### ISSUE TITLE
do you have material/code how find hyperparameters  GBMs, like lightgbm or xgboost or catboos?

### ISSUE THREAD

**Issue by Sandy4321** [gkunapuli/ensemble-methods-notebooks#2]
do you have material/code how find hyperparameters  GBMs, like lightgbm or xgboost or catboos?
meaning what do to step by step how to find hyperparameters ?

**Comment by Sandy4321** [comment]
for example
https://forecastegy.com/posts/catboost-hyperparameter-tuning-guide-with-optuna/

**Comment by Sandy4321** [comment]
I bought your paper book from amazon

**Comment by gkunapuli** [comment]
Yes! Check out [Section 5.4](https://github.com/gkunapuli/ensemble-methods-notebooks/blob/master/Ch5.3and5.4-practical-boosting-with-lightgbm.ipynb) for LightGBM, [Section 6.4](https://github.com/gkunapuli/ensemble-methods-notebooks/blob/master/Ch6.3and6.4-practical-boosting-with-XGBoost.ipynb) for XGBoost and Section 8.2.3 for CatBoost. In addition, the Case Studies for Chapters 5, 6, and 8 also contain examples of hyper-parameter selection. 

Specifically, there are examples of how to perform cross validation for selecting learning rate & ensemble size along with early stopping for efficiency.
